### ISSUE TITLE
Parser for scanning qr codes doesn't know when to end reading

### ISSUE THREAD

**Issue by dekeract10** [mssun/passforios#407]
App version: 0.9.0 (21)

# Bug description and how to reproduce

I've used the script You recommended for qr-encoding my .asc files into a .gif. I opened the "PGP Key" > "ASCII-Armor Encrypted Key" menu. After scanning the images generated by the previously mentioned script I noticed that your scanner doesn't detect when the last image is scanned and just loops endlessly.

# Why it happens

My theory is that this happens because your scanner takes only the current image into account when scanning for the "END PGP PUBLIC KEY BLOCK. I've noticed that after splitting the original .asc file the final part carried only a part of this message (i.e. this message was split across multiple files). I'd guess the solution would be to search for this message in the whole text that was scanned instead of searching for it in just the current image. 

The scanner worked correctly when I decreased block size when splitting the .asc file so that the entirety of the "END PGP BLOCK" message was contained within one file.

I also want to thank you for writing this app and providing Free Software on iOS.

**Comment by mssun** [comment]
Hi, thanks for the report. I believe we have fixed this issue in the TestFlight channel.

You can tap this link on iPhone to use the latest beta version: https://testflight.apple.com/join/whK4zUFG

**Comment by dekeract10** [comment]
I have installed the beta version (0.10.0 38) from your link.  The issue is still present.

I have tried too look at your code (I'm a very inexperienced programmer). The issue seems to be in the PGPKeyArmorImportTableViewController.swift on lines 55-59, from my understanding of the code it searches only the current segment for the line "END PGP BLOCK", not the entirety of the text scanned up to that point.
